conda/environments/datafusion-dev.yaml

```diff
@@ -28,7 +28,7 @@ dependencies:
 - pytest
 - toml
 - importlib_metadata
-- python>=3.7,<3.11
+- python>=3.10
```

Comment:
I do want to point out that cudf has a minimum python version of 3.10. That was the reason for this change. Hopefully that isn't a problem for anyone.